### ISSUE TITLE
Q. when should add a space before { [whitespace/braces]

### ISSUE THREAD

**Issue by elbaro** [cpplint/cpplint#143]
```
1. int a{};
2. struct A{int a,b,c;};
3. if(std::tie(x,y,z) == A{6, 7, 8});
```
cpplint complains about the case 3, says it should be `A {6, 7, 8}`.
My colleagues think this is a false positive.
Can you point to the style guide where it mentions about 3?
What's correct spacing for the case 1, 2, 3?

**Comment by tkruse** [comment]
It looks like a false positive to me, too. cpplint does not use a real parser but just regular expressions, so it can easily become confused.

**Comment by tkruse** [comment]
Based on this https://google.github.io/styleguide/cppguide.html#Variable_and_Array_Initialization I believe no whitespace before array initializer.

**Comment by AlexeyDmitriev** [comment]
I suggest to only look at "{"'s right after ")". It'll reduce the number of false positives. And I don't see any new false negatives from the top of my head
